plugins/cassandra/cassandrawu.cpp

```diff
@@ -3195,13 +3213,29 @@ class CCasssandraWorkUnitFactory : public CWorkUnitFactory, implements ICassandr
             case WUStateDebugRunning:
             case WUStateBlocked:
             case WUStateAborting:
-                // MORE - can see if agent still running, and set to failed if it is not
+                SessionId agent = getUnsignedResult(NULL, cass_row_get_column(row, 1));
+                if (agent && checkWorkUnitSession(wuid, state, agent))
```

Comment:
Not sure - compiling for example I don't know that the compiler actually sets a session id while compiling... if sessionID is not set it means session is not being watched, I think.